packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2639,6 +2641,26 @@ function commitRootImpl(
     markCommitStopped();
   }
 
+  // if (enableTransitionTracing) {
+  const prevRootTransitionCallbacks = root.transitionCallbacks;
+  if (prevRootTransitionCallbacks !== null) {
```

Comment:
Maybe this should check `currentPendingTransitionCallback` instead?

For example, if there were multiple roots scheduled, we still only need a single profiling callback.

There should be a test(s) for multiple roots because that can sometimes be tricky to get right.

Do you also have tests that show multiple commits happening before a paint?

Reply:
A simple example of multiple commits in a single paint is something that uses `useLayoutEffect` to re-render based on layout.

Reply:
I actually think that we won't be able to check currentPendingTransitionCallback, because it might not be populated at the end of commit phase (only the passive effects phase). This check is just making sure the callbacks exist so we don't do extra work when the user didn't actually have any callbacks. 

We should only ever have one currentPendingTransitionCallback because we push everything onto it and clear it when we process it.

Reply:
Maybe a different variable like `isPostPaintCallbackScheduled` then. Then it doesn't matter whether the already-scheduled callbacks are on this root or a different root.

Reply:
Hm, I'm not actually sure what you mean. Different roots will have different end times depending on the commit right? So we actually probably likely do want to schedule different callbacks for them.

Reply:
One paint callback

Reply:
One paint callback